plugins/services/src/js/service-configuration/PodStorageConfigSection.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+import ConfigurationMapHeading from '../../../../../src/js/components/ConfigurationMapHeading';
+import ConfigurationMapSection from '../../../../../src/js/components/ConfigurationMapSection';
+import ConfigurationMapTable from '../components/ConfigurationMapTable';
+import ConfigurationVisible from '../components/ConfigurationVisible';
+import ServiceConfigDisplayUtil from '../utils/ServiceConfigDisplayUtil';
+
+module.exports = ({appConfig}) => {
+  let volumes = (appConfig.volumes || []).reduce((memo, volume) => {
+    let volumeInfo = {
+      volume: volume.name,
+      host: volume.host
+    };
+
+    // Fetch all mounts for this volume in the containers
+    let containerMounts = (appConfig.containers || []).reduce(
+      (cmMemo, container) => {
+        return (container.volumeMounts || []).reduce((cmMemo, volumeMount) => {
```

Comment:
Could we do this a little nicer by filtering and then mapping? You have a lot of these nested functions we can write a little more declarative. It is hard to follow what is going on